bson/decoder_test.go

```diff
@@ -235,6 +235,33 @@ func TestDecoderv2(t *testing.T) {
 			t.Fatalf("Decode error mismatch; expected %v, got %v", ErrDecodeToNil, err)
 		}
 	})
+	t.Run("SetDocumentDecodeType", func(t *testing.T) {
+		type someMap map[string]interface{}
+
+		in := make(someMap)
+		in["foo"] = map[string]interface{}{"bar": "baz"}
+		inType := reflect.TypeOf(in).String()
+		inFooType := reflect.TypeOf(in["foo"]).String()
+
+		bytes, err := Marshal(in)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		var bsonOut someMap
+		dc := new(bsoncodec.DecodeContext).SetDocumentDecodeType(reflect.TypeOf(map[string]interface{}{}))
+		dec, _ := NewDecoderWithContext(*dc, bsonrw.NewBSONDocumentReader(bytes))
+		if err := dec.Decode(&bsonOut); err != nil {
+			t.Fatal(err)
+		}
+
+		bsonOutType := reflect.TypeOf(bsonOut).String()
+		assert.Equal(t, inType, bsonOutType, "expected '%s' to equal '%s'", inType, bsonOutType)
```

Comment:
```suggestion
		assert.Equal(t, inType, bsonOutType, "expected %q to equal %q", inType, bsonOutType)
```
Can just use `%q` for the quotes here and below.